packages/protocol/contracts/common/FeeHandlerSeller.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../common/FixidityLib.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "./UsingRegistry.sol";
+import "../common/Initializable.sol";
+
+contract FeeHandlerSeller is Ownable, Initializable, UsingRegistry {
+  using SafeMath for uint256;
+  using FixidityLib for FixidityLib.Fraction;
+
+  mapping(address => uint256) public minimumReports;
+
+  event MinimumReportsSet(address tokenAddress, uint256 minimumReports);
+  event TokenSold(address tokenAddress, uint256 amount);
+
+  function initialize(
+    address _registryAddress,
+    address[] calldata tokenAddresses,
+    uint256[] calldata newMininumReports
+  ) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(_registryAddress);
+
+    for (uint256 i = 0; i < tokenAddresses.length; i++) {
+      _setMinimumReports(tokenAddresses[i], newMininumReports[i]);
+    }
+  }
+
+  /**
+    * @notice Allows owner to set the minimum number of reports required
+    * @param newMininumReports The new update minimum number of reports required
+    */
+  function setMinimumReports(address tokenAddress, uint256 newMininumReports) public onlyOwner {
+    _setMinimumReports(tokenAddress, newMininumReports);
+  }
+
+  function _setMinimumReports(address tokenAddress, uint256 newMininumReports) internal {
```

Comment:
should this have natspec?

Reply:
I added the natspec to the public functions only, otherwise I'd be copy&pasting.